### PR TITLE
[CRT-1438] Manually Extract Duration When None Provided By Metadata

### PR DESCRIPTION
**Pull request by Kyrluckechuck**
## Description
When parsing some videos that are poorly encoded, such as those generated by the browser MediaRecorder API, they will not have proper metadata such as duration. This means, at present, that it will default to "0" seconds for those videos, which will stop this from erroring out but leads to unintended behaviours with it's usage such as videos which are not encoded properly being marked as "successful" as long as their duration is > 0 seconds.

### CODE REVIEW

File: lib/ffmpeg/movie.rb [hunk 1]
```diff
@@ -150,6 +152,24 @@ def initialize(paths, analyzeduration = 15000000, probesize=15000000 )
...
     end
 
+    # Run null encoding output to get actual duration if none provided
+    def manually_extract_duration
+      command = "#{ffmpeg_command} -i #{@paths.first} -v quiet -stats -f null -"
+      spawn = POSIX::Spawn::Child.new(command)
+
+      # outputs to std error
+      std_error = spawn.err
+      fix_encoding(std_error)
```
Comment (masterful) [inline lib/ffmpeg/movie.rb:162]: side note: it's not pleasant seeing side-effect functions :sweat_smile:

**Review by masterful — APPROVED** [review]
ahhhhhhh - so this is why we're seeing successful encodings of like ~8 seconds??
(or if it's truncating the last ~3 seconds of video?)

great find!

**Comment by Kyrluckechuck** [comment]
> ahhhhhhh - so this is why we're seeing successful encodings of like ~8 seconds??
(or if it's truncating the last ~3 seconds of video?)

@masterful yes and no -- that's why we're seeing them marked as successful, but not why it's happening, unfortunately 😞

**Comment by masterful** [comment]
fair - but this will prevent them from being marked as successful - so hopefully stopping those sorts of errors?

**Pull request merged**